neuralpp/test/quick_tests/symoblic/expression_test.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+
+from neuralpp.symbolic.function import Add
+from neuralpp.symbolic.basic_expression import BasicFunctionApplication, BasicConstant, BasicVariable
+    
+
+def test_basic_constant():
+    # Constant can be anything
+    constant_one = BasicConstant(1)
+    constant_abc = BasicConstant("abc")
+    assert constant_one != constant_abc
+    assert constant_one == BasicConstant(2-1)
+    assert constant_one.subexpression() == []
+    assert not constant_one.contains(constant_abc)
+    assert not constant_one.contains(constant_one)  # A constant does not contains() itself
+    assert constant_one.replace(constant_one, constant_abc) == constant_one  # Any replace() has no effect on constant
+    with pytest.raises(IndexError):
+        constant_one.set(1, constant_abc)  # Any replace() has no effect on constant
+
+
+def test_basic_variable():
+    # Variable is initialized by a variable name.
+    variable_x = BasicVariable("x")
+    variable_y = BasicVariable("y")
+    assert variable_x != variable_y
+    assert variable_x == BasicVariable("x")
+    assert variable_x.subexpression() == []
+    assert not variable_x.contains(variable_y)
+    assert not variable_x.contains(variable_x)  # A variable does not contains() itself
+    assert variable_x.replace(variable_x, variable_y) == variable_x  # Any replace() has no effect on variable
```

Comment:
Again here, I think replacing x by y in x should return y.

Reply:
yes.